### ISSUE TITLE
Virtual Source/Sink Regression

### ISSUE THREAD

**Issue by willcode** [gnuradio/gnuradio#7013]
### What happened?

When a flowgraph using a virtual source/sink in message domain is reloaded, the virtual source/sink ports are taken to be stream domain rather than message domain. This causes a domain mismatch error in the flowgraph.
```
No connection known between domains "stream" and "message"
No connection known between domains "message" and "stream"
```
The problem was most likely introduced in #6889 / #6977

Reported by Jason Merlo on gnuradio-discuss.

### System Information

From maint-3.10 source (v3.10.9.0-rc1)

### GNU Radio Version

3.10 (maint-3.10)

### Specific Version

3.10.9.0-rc1

### Steps to Reproduce the Problem

[message_test.grc.txt](https://github.com/gnuradio/gnuradio/files/13708163/message_test.grc.txt)

**Comment by mbr0wn** [comment]
Yup, I see it too. Now if only I knew how those virtual source/sinks work...
